subprojects/core/src/main/java/org/gradle/api/internal/tasks/options/FieldOptionElement.java

```diff
@@ -139,6 +149,23 @@ public void setValue(Object target, Object value) {
             Property<Object> property = Cast.uncheckedNonnullCast(JavaMethod.of(Object.class, getter).invoke(target));
             property.set(value);
         }
+
+        protected Method getGetter() {
+            return getter;
+        }
+    }
+
+    @NonNullApi
+    private static class MultipleValuePropertyFieldSetter extends PropertyFieldSetter {
+        public MultipleValuePropertyFieldSetter(Method getter, Field field) {
+            super(getter, field);
+        }
+
+        @Override
+        public void setValue(Object target, Object value) {
+            HasMultipleValues<Object> property = Cast.uncheckedNonnullCast(JavaMethod.of(Object.class, getGetter()).invoke(target));
```

Comment:
❓ Is this non-null? I think we currently support setting `null` to properties.

Reply:
This is the same cast as used in the existing [PropertyFieldSetter::setValue](https://github.com/gradle/gradle/blob/master/subprojects/core/src/main/java/org/gradle/api/internal/tasks/options/FieldOptionElement.java#L139) so I think it does not support `null`.